### ISSUE TITLE
Implement blocking publisher API

### ISSUE THREAD

**Issue by SPodjasek** [inway/mojo-rabbitmq-client#17]
Now when we use Promises I believe it would be very convinient to implement default publisher as blocking, when you can enforce non-blocking by using `_p` methods.
This change would allow us to make one-line publishing scripts:

```perl
Mojo::RabbitMQ::Client->publisher(url => 'amqp://.../?exchange=...&routing_key=test')
  ->publish('plain text');
```

**Comment by namotco** [comment]
I second the notion, can you describe what specifically you need help with?

**Comment by SPodjasek** [comment]
I've just published some draft implementation in 5a7cf91 but it requires some code review, docs & tests.

**Comment by SPodjasek** [comment]
Blocking behavior can be acquired with Promise->wait
